Linphone/App.xaml.cs

```diff
@@ -227,7 +239,24 @@ protected override void OnActivated(IActivatedEventArgs args)
                 var arguments = toastArgs.Argument;
                 Initialize(args, arguments);
             }
+        }
 
+        private async void App_CloseRequested(object sender, SystemNavigationCloseRequestedPreviewEventArgs e)
+        {
+            // Todo: Unregister before shutting down.
+            if (CloseApp) return;
+
+            _logger.Information("App_CloseRequested raised omg !!");
+            
+            e.Handled = true;
+            
+            CloseApp = true;
+            
+            if (CallFlowControl.Instance.AgentProfile.IsLoggedIn)
+                await CallFlowControl.Instance.UpdateAgentStatusAsync(BelledonneCommunications.Linphone.Presentation.Dto.AgentStatus.Offline);

```

Comment:
If you make an asynchronous call within your handler, control returns immediately from that asynchronous call. That means that execution can then return from your event handler and your app will move to the next state even though the asynchronous call hasn't completed yet. Use the [GetDeferral](https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.suspendingoperation.getdeferral) method on the [EnteredBackgroundEventArgs](https://docs.microsoft.com/en-us/uwp/api/Windows.ApplicationModel) object that is passed to your event handler to delay suspension until after you call the [Complete](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.deferral.complete) method on the returned [Windows.Foundation.Deferral](https://docs.microsoft.com/en-us/uwp/api/windows.foundation.deferral) object.

Reply:
Resolved.